system/Model.php

```diff
@@ -348,6 +348,41 @@ public function find($id = null)
 		return $row['data'];
 	}
 
+	/**
+	 * Fetches the column of database from $this->table with a primary key
+	 * matching $id.
+	 *
+	 * @param mixed|array|null $id One primary key or an array of primary keys
+	 * @param mixed|array|null $column_name Column name or null for primaryKey
+	 *
+	 * @return array|null array of table's column values
+	 */
+	public function findColumn($id = null, string $columnName = null) : ?array
+	{
+		$columnName = $columnName ?? $this->primaryKey;
+
+		if(!is_string($columnName))
+		{
+			throw DataException::forFindColumnIsNotAString();
+		}
+
+		$tmp = $this->select($columnName)->asArray()->find($id);
+
+		if(count($tmp))
```

Comment:
can be faster with truthy check: `if ($tmp)`

Reply:
I didn't see any benchmarks for this. thx for letting me know.

I made some tests:

```
<?php //7.2.4
    
    $i1 = 8000000;
    
    $d = $tmp = null;
    $a = 'aa';
    
    var_dump($d);
    
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if(!$d) { $tmp = $a;  }
    }
    print 'if(!$d) - do not want to use it : ' . (microtime(true) - $t) . PHP_EOL;
    
    
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if($d == null) { $tmp = $a; }
    }
    print 'd == null - option 1a: ' . (microtime(true) - $t) . PHP_EOL;
        
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if(null == $d) { $tmp = $a; }
    }
    print 'null == d - option 1b: ' . (microtime(true) - $t) . PHP_EOL;
    
    
        
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if($d === null) { $tmp = $a; }
    }
    print '$d === null - option 1c: ' . (microtime(true) - $t) . PHP_EOL;
        
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if(null === $d) { $tmp = $a; }
    }
    print 'null === $d - option 1d: ' . (microtime(true) - $t) . PHP_EOL;
    
    
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        if(is_null($d)) { $tmp = $a; }
    }
    print  'is_null - option 2: ' . (microtime(true) - $t) . PHP_EOL;
    
    $t = microtime(true);
    for($i = 0; $i < $i1; $i++)
    {
        $tmp = $d ?? $a;
    }
    print '?? - option 3: ' . (microtime(true) - $t) . PHP_EOL;
```

```
if(!$d) - do not want to use it : 0.15235209465027
d == null - option 1a: 0.21860098838806
null == d - option 1b: 0.22835683822632
$d === null - option 1c: 0.18482804298401
null === $d - option 1d: 0.15287089347839
is_null - option 2: 0.15867710113525
?? - option 3: 0.15346217155457

if(!$d) - do not want to use it : 0.15974879264832
d == null - option 1a: 0.23204493522644
null == d - option 1b: 0.25987100601196
$d === null - option 1c: 0.15756487846375
null === $d - option 1d: 0.14941382408142
is_null - option 2: 0.14943194389343
?? - option 3: 0.15713381767273

if(!$d) - do not want to use it : 0.14737105369568
d == null - option 1a: 0.20844984054565
null == d - option 1b: 0.21335792541504
$d === null - option 1c: 0.15279507637024
null === $d - option 1d: 0.14113306999207
is_null - option 2: 0.14232611656189
?? - option 3: 0.14513802528381

```

IMHO ?? looks ok, but let's say is_null is slightly faster.

Reply:
As a complete aside - there is the new [Iterator](https://codeigniter4.github.io/CodeIgniter4/testing/benchmark.html#using-the-iterator) that can help you make those tests, too....

Reply:
For example, to test the truthy vs count, I whipped up the following: 

```
$tests = new Iterator();

$array = [1,2,3,4,5,6,7,8,9,10];

$tests->add('count', function() use($array) {
	if (count($array)){}
});

$tests->add('truthy', function() use($array) {
	if ($array){}
});

echo $tests->run(10000);
```

which gave the following output:

```

Test | Time | Memory
-- | -- | --
count | 0.0018 | 0 Bytes
truthy | 0.0016 | 0 Bytes
```

The time varied a bit but on average they were really damn close, with the truthy check being slightly faster. But let's be a little realistic here: I had to run that check **10,000** times to be able to measure the difference, which makes the difference between the so slight as to not even be noticeable during any application I can think of.